pkg/flow/internal/controller/config_export.go

```diff
@@ -0,0 +1,82 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/pkg/river/ast"
+	"github.com/grafana/agent/pkg/river/diag"
+	"github.com/grafana/agent/pkg/river/vm"
+)
+
+type ExportConfigNode struct {
+	configNode      sharedBlockNode
+	onExportsChange func(exports map[string]any) // Invoked when the managed component updated its exports
+}
+
+var _ BlockNode = (*ExportConfigNode)(nil)
+
+// NewExportConfigNode creates a new ExportConfigNode from an initial ast.BlockStmt.
+// The underlying config isn't applied until Evaluate is called.
+func NewExportConfigNode(block *ast.BlockStmt, globals ComponentGlobals, isInModule bool) (*ExportConfigNode, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	if !isInModule {
+		diags.Add(diag.Diagnostic{
+			Severity: diag.SeverityLevelError,
+			Message:  "export blocks only allowed inside a module",
+			StartPos: ast.StartPos(block).Position(),
+			EndPos:   ast.EndPos(block).Position(),
+		})
+	}
+
+	return &ExportConfigNode{
+		configNode: sharedBlockNode{
+			label:         block.Label,
+			nodeID:        BlockComponentID(block).String(),
+			componentName: block.GetBlockName(),
```

Comment:
`componentName` isn't right here though, since config blocks aren't components. 

I'm starting to lean a little more firmly against the idea of the sharedBlockNode struct and prefer copying the fields, but I'd like a second opinion from @tpaschalis to see if I'm overthinking it.

Reply:
I guess it would just become `name` in that case. That being said, if it's a little too object oriented of an approach for normal golang patterns and not a pattern we typically use I'm happy to nuke it once @tpaschalis takes a quick look.

Reply:
This should be all set.

Reply:
Hey team, I'll look at it tomorrow if that's okay.